test/expint.jl

```diff
@@ -0,0 +1,61 @@
+using Base.MathConstants
+
+@testset "exponential integral" begin
+    @testset "E₁" begin
+        @test E₁(0) == Inf
+        @test E₁(1) ≅ 0.219383934395520273677163775460121649031047293406908207577
+        @test E₁(2) ≅ 0.048900510708061119567239835228049522314492184963023116327
+        @test E₁(10) ≅ 4.1569689296853242774028598102781803843462900824195331e-6
+        @test E₁(100) ≅ 3.6835977616820321802351926205081189876552201369095676e-46
+        @test E₁(500) ≅ 1.422076782253638422098193936057278281607864840846107e-220
+        @test E₁(800) ≅ 0.0
```

Comment:
Need to test each of the intervals of different rational approximants, to make sure that every branch is tested.

Reply:
especially for the complex case — test right after the crossover point to a new approximant